cmd/ui/src/test-utils.jsx

```diff
@@ -27,75 +27,103 @@ import createSagaMiddleware from 'redux-saga';
+                </StyledEngineProvider>
+            </QueryClientProvider>
+        </Provider>    
+    )
+}
+
 const customRender = (
     ui,
     {
         initialState = {},
-        queryClient = new QueryClient({
-            defaultOptions: {
-                queries: {
-                    retry: false,
-                },
-            },
-        }),
+        queryClient = createDefaultQueryClient(),
         history = createMemoryHistory(),
-        theme = createTheme({
-            palette: {
-                primary: {
-                    main: '#406f8e',
-                    light: '#709dbe',
-                    dark: '#064460',
-                    contrastText: '#ffffff',
-                },
-                neutral: {
-                    main: '#e0e0e0',
-                    light: '#ffffff',
-                    dark: '#cccccc',
-                    contrastText: '#000000',
-                },
-                background: {
-                    paper: '#fafafa',
-                    default: '#e4e9eb',
-                },
-                low: 'rgb(255, 195, 15)',
-                moderate: 'rgb(255, 97, 66)',
-                high: 'rgb(205, 0, 117)',
-                critical: 'rgb(76, 29, 143)',
-            },
-        }),
-        store = configureStore({
-            reducer: rootReducer,
-            preloadedState: initialState,
-            middleware: (getDefaultMiddleware) => {
-                return [...getDefaultMiddleware({ serializableCheck: false }), createSagaMiddleware()];
-            },
-        }),
+        theme = createTheme(defaultTheme),
+        store = createDefaultStore(initialState),
         ...renderOptions
     } = {}
 ) => {
-    const AllTheProviders = ({ children }) => {
-        return (
-            <Provider store={store}>
-                <QueryClientProvider client={queryClient}>
-                    <StyledEngineProvider injectFirst>
-                        <ThemeProvider theme={theme}>
-                            <NotificationsProvider>
-                                <CssBaseline />
-                                <Router location={history.location} navigator={history}>
-                                    <SnackbarProvider>{children}</SnackbarProvider>
-                                </Router>
-                            </NotificationsProvider>
-                        </ThemeProvider>
-                    </StyledEngineProvider>
-                </QueryClientProvider>
-            </Provider>
-        );
-    };
+    const AllTheProviders = ({ children }) => createProviders({ queryClient, history, theme, store, children });
     return render(ui, { wrapper: AllTheProviders, ...renderOptions });
 };
 
+const customRenderHook = (
+    hook,
+    {
+        initialState = {},
+        queryClient = createDefaultQueryClient(),
+        history = createMemoryHistory(),
+        theme = createTheme(defaultTheme),
+        store = createDefaultStore(initialState),
```

Comment:
I never thought about using a param as the default value of another param. very neat!